ironfish-rust/src/keys/proof_authorizing_key.rs

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use jubjub::Fr;
+
+use crate::{
+    errors::{IronfishError, IronfishErrorKind},
+    serializing::{bytes_to_hex, hex_to_bytes},
+};
+
+pub type ProofAuthorizingKey = jubjub::Fr;
+
```

Comment:
Is there a better way to do this/ better location for this?

Reply:
i could do ProofAuthorizingKey { nsk: jubjub::Fr } but i think it unnecessarily complicates things

Reply:
It seems like the purpose of this is to be able to call `proof_authorizing_key.hex_key()`?

I'm a Rust novice, but it seems like it might be simpler to add something like this to our `serializing` mod:
```
pub trait HexKey {
    fn hex_key(&self) -> String;
}

impl HexKey for jubjub::Fr {
    fn hex_key(&self) -> String {
        bytes_to_hex(&self.to_bytes())
    }
}
```

Reply:
You could also call `bytes_to_hex(proof_authorizing_key.to_bytes())` directly

Reply:
That is 1 reason + having `ProofAuthorizingKey` as the type everywhere would make it easier to read. Otherwise you have to do some mental math to confirm indeed that `jubjub::Fr` is a proof_authorizing_key by looking at the Sapling key. Don't feel strongly about this thought. A trait would be useful regardless

Reply:
The type alias and trait for `ProofAuthorizingKey` seem unnecessary to me for now

Reply:
Ok I'll change it to just the `bytes_to_hex(proof_authorizing_key.to_bytes())`

Reply:
I tried the above and I think I still prefer to have it as a separate type/ alias. The code is much cleaner, fits the patterns with everything else, and doesn't require and additional dependency to be added to the rust-nodejs project